tests/test_from_thread.py

```diff
@@ -187,6 +187,31 @@ def external_thread() -> None:
         for thread_id in thread_ids:
             assert thread_id == threading.get_ident()
 
+    async def test_successful_call_non_coro(self) -> None:
+        class AsyncGetThreadId:
+            def __init__(self) -> None:
+                self.thread_id = threading.get_ident()
+
+            def __await__(self) -> Iterator[int]:
+                assert self.thread_id == threading.get_ident()
+
+                async def coro() -> int:
+                    return self.thread_id
+                return coro().__await__()
```

Comment:
To be honest this is a bit lower-level into the asyncio world than I am familiar with, so this is the best I could come up with that made sense and worked, but I'm open to suggestions on how this can be made simpler w/o the closure